### PR TITLE
impl Contains for all geometries

### PR DESCRIPTION
**Pull request by michaelkirk**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Based on the test suite and correctness fixes in #882 so please review that first.

Fixes https://github.com/georust/geo/issues/535

### CODE REVIEW

**Comment by michaelkirk** [comment]
I found some correctness issues with this while adding more of the JTS test suite, so I'm going to close for now. 

(I'm actively working on it, so hopefully it won't be long before I re-open)

**Comment by michaelkirk** [comment]
> I found some correctness issues with this while adding more of the JTS test suite, so I'm going to close for now.


Ok, I fixed these in #882, so once 882 is approved, this is ready for review.

File: geo/src/algorithm/mod.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ pub use area::Area;
 pub mod bearing;
 pub use bearing::Bearing;
 
+/// Boolean Ops such as union, xor, difference;
```
Comment (michaelkirk) [inline geo/src/algorithm/mod.rs:13]: Trying to keep this file alphabetically organized.

**Comment by michaelkirk** [comment]
#882 has been merged! So this is ready for review.

**Review by rmanoka — APPROVED** [review]
lgtm!

**Comment by michaelkirk** [comment]
bors r=rmanoka

**Pull request merged**